train_concept_identification.py

```diff
@@ -303,6 +351,131 @@ def train_model(model: nn.Module,
     eval_writer.add_scalar('loss', dev_loss, epoch)
     eval_writer.add_scalar('f-score', fscore, epoch)
     train_writer.add_scalar('f-score', batch_f_score_train, epoch)
+    # Use scheduler for LR optimization
+    if scheduler:
+      scheduler.step()
+
+  # Save pretrained model
+  if config.SAVE_PATH:
+    torch.save(model.state_dict(), config.SAVE_PATH)
+    
+
+def load_model_weights(model, load_path: str, component: str):
+  """
+    Loads pretrained model weights to current model
+    Args:
+      model: model whose weights need to be updated
+      load_path (str): file path to load from
+      component (str): part of pretrained model to load
+  """
+  if(os.path.exists(load_path)):
+    pretrained_dict = torch.load(load_path)
+    # If we're not loading a specific component, we're loading everything
+    if component:
+      # Select weights belonging to component
+      pretrained_dict = {k: v for k, v in pretrained_dict.items() if component in k}
+    model_dict = model.state_dict()
+    model_dict.update(pretrained_dict)
+    model.load_state_dict(model_dict)
+  return model
+
+
+def generate_sentences(len: int):
```

Comment:
this should not be done in the train script -- rather should somehow be done in the dataset